### PR TITLE
Bug fixes and UI updates for datasources

### PR DESCRIPTION
**Pull request by sejli**
### Description
* Adds try catch for fetching tables
* Updates `Accelerations` columns to `Associations` for better visibility
  * Skipping indices and covering indices have their associated table listed now

### Issues Resolved
N/A

### Check List
- [ ] New functionality includes testing.
  - [ ] All tests pass, including unit test, integration test and doctest
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
  - [ ] New functionality has user manual doc added
- [x] Commits are signed per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Review by Swiddis — APPROVED** [review]

File: public/framework/catalog_cache/cache_loader.tsx [hunk 1]
```diff
@@ -164,12 +164,14 @@ export const updateTableColumnsToCache = (
     ts.name === tableName ? { ...ts, columns: tableColumns } : { ...ts }
   );
 
-  CatalogCacheManager.updateDatabase(dataSourceName, {
-    ...cachedDatabase,
-    tables: newTables,
-    lastUpdated: currentTime,
-    status: CachedDataSourceStatus.Updated,
-  });
+  if (cachedDatabase.status === CachedDataSourceStatus.Updated) {
```
Comment (ps48) [inline public/framework/catalog_cache/cache_loader.tsx:167]: Can remove this line have a more covered checked on my PR: https://github.com/opensearch-project/dashboards-observability/pull/1617

File: public/components/datasources/components/manage/associated_objects/modules/associated_objects_table.tsx [hunk 2]
```diff
@@ -196,10 +198,9 @@ export const AssociatedObjectsTable = (props: AssociatedObjectsTableProps) => {
       return clauses.some((clause) => {
         if (clause.field !== ASSC_OBJ_TABLE_ACC_COLUMN_NAME) {
           return associatedObject[clause.field] === clause.value;
-        } else if (
-          clause.field === ASSC_OBJ_TABLE_ACC_COLUMN_NAME &&
-          Array.isArray(associatedObject.accelerations)
-        ) {
+        } else if (clause.field === ASSC_OBJ_TABLE_ACC_COLUMN_NAME) {
+          console.log('in here');
+          console.log(associatedObject);
```
Comment (ps48) [inline public/components/datasources/components/manage/associated_objects/modules/associated_objects_table.tsx:203]: Let's remove console logs

File: public/components/datasources/components/manage/associated_objects/associated_objects_tab.tsx [hunk 3]
```diff
@@ -205,7 +207,13 @@ export const AssociatedObjectsTab: React.FC<AssociatedObjectsTabProps> = (props)
   // Load tables and accelerations if empty or retrieve from cache if not
   useEffect(() => {
     if (datasource.name && selectedDatabase) {
-      const databaseCache = CatalogCacheManager.getDatabase(datasource.name, selectedDatabase);
+      let databaseCache;
+      try {
+        databaseCache = CatalogCacheManager.getOrCreateDataSource(datasource.name);
+      } catch (error) {
+        setToast('Your cache is outdated, refresh databases and tables', 'warning');
```
Comment (ps48) [inline public/components/datasources/components/manage/associated_objects/associated_objects_tab.tsx:214]: Can you please add `console.error`

File: public/components/datasources/components/manage/associated_objects/associated_objects_tab.tsx [hunk 4]
```diff
@@ -237,7 +245,12 @@ export const AssociatedObjectsTab: React.FC<AssociatedObjectsTabProps> = (props)
   useEffect(() => {
     if (datasource.name && selectedDatabase) {
       const tablesStatus = tablesLoadStatus.toLowerCase();
-      const databaseCache = CatalogCacheManager.getDatabase(datasource.name, selectedDatabase);
+      let databaseCache;
+      try {
+        databaseCache = CatalogCacheManager.getDatabase(datasource.name, selectedDatabase);
+      } catch (error) {
+        return;
```
[the remark this answers is not in the document]
Comment (ps48) [reply public/components/datasources/components/manage/associated_objects/associated_objects_tab.tsx:252]: Do we need a toast message here?

File: public/components/datasources/components/manage/associated_objects/associated_objects_details_flyout.tsx [hunk 5]
```diff
@@ -223,12 +225,17 @@ export const AssociatedObjectsDetailsFlyout = ({
 
   useEffect(() => {
     if (loadStatus.toLowerCase() === DirectQueryLoadingStatus.SUCCESS) {
```
Comment (ps48) [inline public/components/datasources/components/manage/associated_objects/associated_objects_details_flyout.tsx:227]: Let's handle the failed case where to stop loading the table.

Comment (sejli) [reply public/components/datasources/components/manage/associated_objects/associated_objects_details_flyout.tsx:226]: Loading is handled by `loadStatus`. If it fails, it'll be handled already.

**Review by ps48 — APPROVED** [review]

**Pull request merged**